repos/application-insights.proj

```diff
@@ -1,6 +1,12 @@
 <?xml version="1.0" encoding="utf-8"?>
 <Project ToolsVersion="15.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <PropertyGroup>
+    <PreferredHash>53b80940842204f78708a538628288ff5d741a1d</PreferredHash>
```

Comment:
Just so I understand.  To move a repo forward, do each of these PreferredHashes need to be updated?  Are these auto-updated in some way?

Reply:
Correct, the `PreferredHash` is what actually has to move forward.  Currently this will look like:
1. One of our submodules updates.
2. We get an auto-PR that changes `eng/Version.Details.xml` and nothing else.
3. CI build fails.
4. Someone updates `PreferredHash`.
5. CI build hopefully passes and is merged.

Using the version numbers and hashes out of `eng/Version.Details.xml` for our build directly is part of what I was planning for phase 2 of the dependency changes.